sparse/coo.py

```diff
@@ -2006,7 +2014,39 @@ def astype(self, dtype, out=None):
         assert out is None
         return self.elemwise(np.ndarray.astype, dtype)
 
-    def maybe_densify(self, max_size=1000, min_density=0.25):
+    @contextmanager
+    def configure_densification(self, densify, max_size=None, min_density=None):
+        if isinstance(self._densification_config, DensificationConfig):
+            old_densification_config = DensificationConfig(self._densification_config)
+            self._densification_config.make_copy_of(DensificationConfig(densify, max_size, min_density))
+        else:
+            old_densification_config = self._densification_config
+            self._densification_config = DensificationConfig(densify, max_size, min_density)
+
+        yield
+
+        if isinstance(old_densification_config, DensificationConfig):
+            self._densification_config.make_copy_of(old_densification_config)
+        else:
+            self._densification_config = old_densification_config
+
+    @property
+    def densification_config(self):
```

Comment:
I don't like how `COO` knows how to deal with all the different `DensificationConfig` flavors. All of this should be entirely encapsulated in `DensificationConfig`.

Reply:
Done.